linux/deploy_vm/add_vm_virtual_tpm.yml

```diff
@@ -0,0 +1,51 @@
+# Copyright 2022 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# The prerequsite of adding virtual TPM device to VM is
+# key provider configured on vCenter.
+#
+# When 'key_provider_type' parameter is set, add a new key
+# provider with specified type firstly, now only support
+# Standard key provider configuration.
+#
+- block:
+    - name: Check key provider type value
+      assert:
+        that:
+          - key_provider_type | lower in ['standard', 'native']
```

Comment:
I think we can move `linux/deploy_vm/add_vm_virtual_tpm.yml` to `common/vm_add_virtual_tpm.yml` as it doesn't have linux specific operations. And if `native` key provider is not supported at now, we can only check `key_provider_type | lower == 'standard'`. After `native` key provider is supported, we can add it back here.

Reply:
Moved task file to common folder.